src/Utils.js

```diff
@@ -80,7 +81,12 @@ export function nodeEqual(a, b) {
       return false;
     }
   }
-  return leftKeys.length === Object.keys(right).length;
+
+  if (typeof a !== 'string' && typeof a !== 'number') {
+    return leftKeys.length === Object.keys(right).length;
```

Comment:
I think it's pretty safe to assume it's available.

`Object.keys` has really wide support. On the browser side you'd have to use less than IE9, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys

On node side, it's been included since at least 0.10. I'm not sure exactly when it was added to V8, but I think being in 0.10 makes it more than safe to use natively